src/arrayMethodSort.js

```diff
@@ -4,8 +4,36 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
+  [].__proto__.sort2 = function (compareFunction) {
     // write code here
+    let tempFunction = compareFunction;
+
+    if (typeof compareFunction !== 'function') {
+      tempFunction = function (a, b) {
+        const strA = String(a);
+        const strB = String(b);
+
+        if (strA < strB) {
+          return -1;
+        }
+
+        if (strA > strB) {
+          return 1;
+        }
+
+        return 0;
+      };
+    }
```

Comment:
The check for 'compareFunction' being a function is correct. However, the default comparator could be simplified by directly comparing strings without assigning them to variables 'strA' and 'strB'.